### PR TITLE
Dropdown issues

### LINKED ISSUE #432: Dropdown issues
### `canClear` does not exist

This is not exposed as `can-clear` but `show-clear-button`. The same for the JS property: `showClearButton`.

### `selectedOption` does not work

The property is set but it is not rendered, so at the end is not really selected.

```html
<as-dropdown></as-dropdown>
<script>
  const dropdown = document.querySelector('as-dropdown');
  dropdown.options = [
    { text: 'All', value: 'all' },
    { text: 'Open', value: 'open' },
    { text: 'Unfulfilled', value: 'unfulfilled' },
    { text: 'Unpaid', value: 'unpaid' }
  ];
  dropdown.selectedOption = 'all'; // It does not work
</script>
```

This is a workaround:

```html
<as-dropdown></as-dropdown>
<script>
  const dropdown = document.querySelector('as-dropdown');
  dropdown.options = [
    { text: 'All', value: 'all' },
    { text: 'Open', value: 'open' },
    { text: 'Unfulfilled', value: 'unfulfilled' },
    { text: 'Unpaid', value: 'unpaid' }
  ];
  dropdown.componentOnReady().then(() => {
    dropdown.selectedOption = 'all'; // It works!
  });
</script>
```

But you need to be careful because if the property is already set, the workaround does not work because the `selectedOption` value is not changing:

```html
<as-dropdown></as-dropdown>
<script>
  const dropdown = document.querySelector('as-dropdown');
  dropdown.options = [
    { text: 'All', value: 'all' },
    { text: 'Open', value: 'open' },
    { text: 'Unfulfilled', value: 'unfulfilled' },
    { text: 'Unpaid', value: 'unpaid' }
  ];
  dropdown.selectedOption = 'all';
  dropdown.componentOnReady().then(() => {
    dropdown.selectedOption = 'all'; // It does not work!
  });
</script>
```

### list menu is in the back

selection_359

This is a workaround:

```
.as-dropdown__list {
  z-index: 1
}
```

selection_360

### PR DESCRIPTION
**Pull request by ivanmalagon**
Fixes #432 
---

WIP

### CODE REVIEW

File: packages/components/src/components/as-dropdown/as-dropdown.tsx [hunk 1]
```diff
@@ -134,18 +132,32 @@ export class Dropdown {
     return option === this.selectedOptionObject;
   }
 
-  private select(option) {
-    if (this.selectedOption === option) {
+  private select(optionObject) {
```
Comment (IagoLast) [inline packages/components/src/components/as-dropdown/as-dropdown.tsx]:
```suggestion
  private _select(optionObject) {
```

File: packages/components/src/components/as-dropdown/as-dropdown.tsx [hunk 2]
```diff
@@ -134,18 +132,32 @@ export class Dropdown {
...
-    this.selectedOption = option.value;
-    this.selectedOptionObject = option;
+    this.selectedOption = optionObject.value;
+    this.selectedOptionObject = optionObject;
 
     this.closeList();
     this.emitOption();
   }
 
+  private selectFromValue(value) {
```
Comment (IagoLast) [inline packages/components/src/components/as-dropdown/as-dropdown.tsx]:
```suggestion
  private _selectFromValue(value) {
```

File: packages/components/src/components/as-dropdown/as-dropdown.tsx [hunk 3]
```diff
@@ -78,94 +82,95 @@ export class Dropdown {
...
 
-    this.closeList();
-    this.emitOption();
+    this._closeList();
+    this._emitOption();
   }
 
-  private toggleList() {
+  private _selectFromValue(value) {
+    if (value === undefined || value === null) {
```
Comment (IagoLast) [inline packages/components/src/components/as-dropdown/as-dropdown.tsx]: should we add a tests for the edge case?

Comment (ivanmalagon) [reply packages/components/src/components/as-dropdown/as-dropdown.tsx]: added

**Comment by ivanmalagon** [comment]
docs changed
